gamestonk_terminal/cryptocurrency/due_diligence/glassnode_view.py

```diff
@@ -0,0 +1,200 @@
+    else:
+        _, ax1 = plt.subplots(figsize=(25, 7))
+
+        ax1.fill_between(
+            df_addresses[df_addresses["v"] < 0].index,
+            df_addresses[df_addresses["v"] < 0]["v"].values / 1e3,
+            np.zeros(len(df_addresses[df_addresses["v"] < 0])),
+            facecolor="red",
+        )
+        ax1.fill_between(
+            df_addresses[df_addresses["v"] >= 0].index,
+            df_addresses[df_addresses["v"] >= 0]["v"].values / 1e3,
+            np.zeros(len(df_addresses[df_addresses["v"] >= 0])),
+            facecolor="green",
+        )
+
+        ax1.set_ylabel(
+            f"30d change of {asset} supply held in exchange wallets [thousands]"
+        )
+        plt.title(
+            f"{asset}: Exchange Net Position Change - {'all exchanges' if exchange == 'aggregated' else exchange}"
+        )
+        plt.xlim([df_addresses.index[0], df_addresses.index[-1]])
+        plt.gca().xaxis.set_major_formatter(mdates.DateFormatter("%m-%Y"))
+        plt.gca().xaxis.set_major_locator(mdates.MonthLocator(interval=3))
+        plt.gcf().autofmt_xdate()
+
+        plt.show()
+    print("")
+
+    export_data(
+        export,
+        os.path.dirname(os.path.abspath(__file__)),
+        "change",
+        df_addresses,
+    )
+
+
+def display_exchange_balances(
+    asset: str,
+    exchange: str,
+    since: int,
+    until: int,
+    interval: str,
+    percentage: bool,
+    export: str = "",
+) -> None:
+    """Display total amount of coins held on exchange addresses in units and percentage.
+    [Source: https://glassnode.org]
+
+    Parameters
+    ----------
+    asset : str
+        Asset to search active addresses (e.g., BTC)
+    exchange : str
+        Exchange to check net position change (e.g., binance)
+    since : int
+        Initial date timestamp (e.g., 1_614_556_800)
+    until : int
+        End date timestamp (e.g., 1_614_556_800)
+    interval : str
+        Interval frequency (e.g., 24h)
+    percentage : bool
+        Show percentage instead of stacked value.
+    export : str
+        Export dataframe data to csv,json,xlsx file
+    """
+
+    df_balance = get_exchange_balances(asset, exchange, interval, since, until)
+
+    if df_balance.empty:
+        print("Error in glassnode request")
+    else:
+        _, ax1 = plt.subplots(figsize=(25, 7))
+        if percentage:
+            plt.plot(df_balance.index, df_balance["percentage"] * 100, c="k")
+        else:
+            plt.plot(
+                df_balance.index, df_balance["stacked"] / 1000, c="k"
+            )  # in thousands
+
+        ax2 = ax1.twinx()
+
+        plt.grid()
+        plt.plot(df_balance.index, df_balance["price"], c="orange")
+
+        ax1.set_ylabel(f"{asset} units [{'%' if percentage else 'thousands'}]")
+        ax2.set_ylabel(f"{asset} price [$]", c="orange")
+        plt.title(
+            f"{asset}: Total Balance in {'all exchanges' if exchange == 'aggregated' else exchange}"
+        )
```

Comment:
Honestly not too sure what the difference is, but since you define an axes object with subplots, for consistency it would make sense to use the axes methods (`ax1.plot()`, `ax2.plot()`, `ax1.grid()` etc).  Mixing between ax and plt is a bit confusing

Reply:
The plot just acts on top of the last axes defined.

The reason the definition of axis is needed is to invoke the mirror (twinx)